### DISCUSSION TITLE
Add ONVIF Discovery to Frigate

### DISCUSSION THREAD

**Discussion by bitvivid-deepak** [blakeblackshear/frigate#8422] [General]
Hello to the wonderful community. I am impressed with Frigate and I want to extend the current ONVIF functionality to add discovery but don't know where to start.
Also I want to add some menu items on the Frigate UI but again I am looking for some quick starts to shorten the learning curve.

Any help is highly appreciated.

**Comment by NickM-27** [comment]
I think onvif discovery might make more sense when frigate supports UI configuration (or at least can start without any cameras configured)

https://deploy-preview-6262--frigate-docs.netlify.app/development/contributing is a good place to start. In general I would suggest creating a discussion explaining your ideas before implementing them so no time is wasted on it in the case that it doesn't fit with the projects goals

**Comment by bitvivid-deepak** [comment]
Thank you so much Nick. You have been very helpful.
